### DISCUSSION TITLE
prevent anyone from ldap to connect

### DISCUSSION THREAD

**Discussion by knpshkn** [MarcJHuber/event-driven-servers#37] [Q&A] [answered]
Hello, in tac_plus we had this snippet to prevent anyone that exists in ldap from connecting to the hardware. We specify users manually in the configuration with password = mavis . In tac_plus-ng this trick doesn't work. Maybe you can advise how we can replace group/acl in this case?
```
    host = world {
        address = 0.0.0.0/0
        default group = deny_login
    group = deny_login {
        acl = acl_deny_login
    }

```

**Comment by MarcJHuber** [comment, marked as answer]
Hi,

in tac_plus-ng that functionality did move, more or less, to the ruleset you're using. If your goal is to permit requests that match a certain criteria (like group membership) and deny everything left then a final "deny" after evaluating the positive list is enough. Just handle the positive criteria first, and let the bad one fall through.

ruleset {
&nbsp;    rule {
&nbsp; &nbsp;   &nbsp;       script {
&nbsp; &nbsp;   &nbsp;  &nbsp;          if (member == ...) { profile = ... permit }
&nbsp;  &nbsp;  &nbsp;  &nbsp;          if (member == ...) { profile = ... permit }
&nbsp;  &nbsp;  &nbsp;   &nbsp;         deny
&nbsp;  &nbsp;  &nbsp;       }
&nbsp;  &nbsp;    }
}

(Apologies -- "Add Code" doesn't seem to work, so this isn't well-formatted.)

Cheers,

Marc
